onvif/client.py

```diff
@@ -305,20 +305,23 @@ def get_definition(self, name, portType=None):
 
         return xaddr, wsdlpath, binding_name
 
-    def create_onvif_service(self, name, from_template=True, portType=None):
+    def create_onvif_service(self, name, from_template=True, portType=None, transport=None):
         '''Create ONVIF service client'''
 
         name = name.lower()
         xaddr, wsdl_file, binding_name = self.get_definition(name, portType)
 
         with self.services_lock:
+            if not transport:
+                transport = self.transport
+
             service = ONVIFService(xaddr, self.user, self.passwd,
                                    wsdl_file, self.encrypt,
                                    self.daemon, no_cache=self.no_cache,
                                    portType=portType,
                                    dt_diff=self.dt_diff,
                                    binding_name=binding_name,
-                                   transport=self.transport)
```

Comment:
`self.transport` gets set in the `__init__` method, so you should be able to pass your `AsyncTransport` when calling `ONVIFCamera(transport=my_async_transport)` instead of doing this logic and passing down to the other create services.

Reply:
No, because the ONVIFCamera does service calls itself and is not async-aware.